src/pages/SingleProductPage.js

```diff
@@ -1,22 +1,95 @@
-import React, { useEffect } from 'react'
-import { useParams, useHistory } from 'react-router-dom'
-import { useProductsContext } from '../context/products_context'
-import { single_product_url as url } from '../utils/constants'
-import { formatPrice } from '../utils/helpers'
+import React, { useEffect } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { useProductsContext } from '../context/products_context';
+import { single_product_url as url } from '../utils/constants';
+import { formatPrice } from '../utils/helpers';
 import {
   Loading,
   Error,
   ProductImages,
   AddToCart,
   Stars,
   PageHero,
-} from '../components'
-import styled from 'styled-components'
-import { Link } from 'react-router-dom'
-
+} from '../components';
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
 const SingleProductPage = () => {
-  return <h4>single product page</h4>
-}
+  const { id } = useParams();
+  const idNumber = Number(id); //converted to number
+  console.log(id);
+  //const navigate = useNavigate();
+  const {
+    single_product_loading: loading,
+    single_product_error: error,
+    single_product: product,
+    fetchSingleProduct,
+  } = useProductsContext();
+//but i still cant fetch single product
+  useEffect(() => {
+    fetchSingleProduct(url, idNumber);
+    // eslint-disable-next-line 
+  }, [idNumber]);
+  
+  useEffect(() => {
+    if (error) {
+      setTimeout(() => {
+       // navigate('/');
+      }, 3000);
+    }
+    // eslint-disable-next-line
+  }, [error]);
+  if (loading) {
+    return <Loading />;
+  }
+  if (error) {
+    return <Error />;
+  }
+
+  const {
+    name,
+    price,
+    description,
+    stock,
```

Comment:
does your backend currently have this field in the schema? I think instead you have a field called \`inventory\` that may be able to play the same part here.